java/core/src/main/java/co/worklytics/psoxy/impl/RESTApiSanitizerImpl.java

```diff
@@ -117,12 +121,28 @@ Predicate<Map.Entry<Endpoint, Pattern>> getHasPathRegexMatchingUrl(String relati
 
     @VisibleForTesting
     Predicate<Map.Entry<Endpoint, Pattern>> getHasPathTemplateMatchingUrl(URL url) {
-        return (entry) ->
-                entry.getKey().getPathTemplate() != null && entry.getValue().matcher(url.getPath()).matches()
-                        && allowedQueryParams(entry.getKey(), URLUtils.queryParamNames(url));
+        return (entry) -> {
+            if (entry.getKey().getPathTemplate() != null) {
+                Matcher matcher = entry.getValue().matcher(url.getPath());
+                if (matcher.matches()) {
+                    boolean allParamsValid =
+                            entry.getKey().getPathParameterSchemasOptional()
+                                    .map(schemas -> schemas.entrySet().stream()
+                                        .map(paramSchema -> parameterSchemaUtils.validate(paramSchema.getValue(), matcher.group(paramSchema.getKey())))
+                                         .allMatch(b -> b))
```

Comment:
```suggestion
                                        .allMatch(paramSchema -> parameterSchemaUtils.validate(paramSchema.getValue(), matcher.group(paramSchema.getKey()))))
```
don't need the map, right?